### PR TITLE
fix: removes duplicate auth types

### PR DESCRIPTION
**Pull request by akshbhu**
*Issue #5240 

*Description of changes:*

* Introduce case where no configure additional auth type option selected

* Added test for the same case

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts [hunk 1]
```diff
@@ -448,6 +448,16 @@ async function askAdditionalAuthQuestions(context, authConfig, defaultAuthType)
 
       authConfig.additionalAuthenticationProviders.push(config);
     }
+  } else {
+    const currentAuthConfig = getAppSyncAuthConfig(context.amplify.getProjectMeta());
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:452]: move currentAuthConfig on line 429 out of if block and reuse in if and else

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts [hunk 2]
```diff
@@ -448,6 +448,16 @@ async function askAdditionalAuthQuestions(context, authConfig, defaultAuthType)
...
+    if (
+      !_.isUndefined(currentAuthConfig.additionalAuthenticationProviders) &&
+      !_.isEmpty(currentAuthConfig.additionalAuthenticationProviders)
+    ) {
+      authConfig.additionalAuthenticationProviders = currentAuthConfig.additionalAuthenticationProviders.filter(
+        p => p.authenticationType !== defaultAuthType,
+      );
+    }
```
Comment (edwardfoyle) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:460]: can simplify to
```suggestion
    const {additionalAuthenticationProviders} = getAppSyncAuthConfig(context.amplify.getProjectMeta());
    authConfig.additionalAuthenticationProviders = (additionalAuthenticationProviders || []).filter(
      p => p.authenticationType !== defaultAuthType,
    );
```

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts (hunk 1 above)
Comment (akshbhu) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts]: fixed

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts (hunk 2 above)
Comment (akshbhu) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts]: done!!

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**